src/users.rs

```diff
@@ -0,0 +1,243 @@
+use data_encoding::HEXUPPER;
+use rand;
+use ring::error::Unspecified;
+use ring::{digest, pbkdf2};
+use rusqlite::NO_PARAMS;
+use rusqlite::{named_params, params, Connection};
+use std::error::Error;
+use std::fmt;
+
+// User Struct
+pub struct User {
+    pub name: String,
+    pub salt: String,
+    pub hash: String,
+    pub active: bool,
+}
+
+// Constants
+const CREDENTIAL_LEN: usize = digest::SHA512_OUTPUT_LEN;
+const N_ITER: u32 = 100_000;
+
+// Error Used for User related Functions
+// Implements the traits of Error
+#[derive(Debug)]
+struct UserError(String);
```

Comment:
I'm not convinced we need this, we can probably just return the existing errors. I've been using the `failure` crate, which we can continue to use for the minute, but there are better options now.

Reply:
Will remove